src/collector.cpp

```diff
@@ -200,14 +212,15 @@ namespace { namespace aux {
 
 			// commit if everything is ok
 			ai_list_ = std::move(ai_list);
-			ai_      = ai;
 		}
 
 		fd_handle_t try_bind_to_addr(os_addrinfo_t *ai)
 		{
 			fd_handle_t fd { os_unix::socket_ex(ai->ai_family, ai->ai_socktype, ai->ai_protocol) };
 			os_unix::setsockopt_ex(*fd, SOL_SOCKET, SO_REUSEADDR, 1);
 			os_unix::setsockopt_ex(*fd, SOL_SOCKET, SO_REUSEPORT, 1);
+			if (ai->ai_family == AF_INET6)
+				os_unix::setsockopt_ex(*fd, IPPROTO_IPV6, IPV6_V6ONLY, int(~0));
```

Comment:
Why ipv6 only and why ~0 and not 1, let's say?

Reply:
this option is ipv6 only, so "if" is for that :)

About the option itself: when host is "*" it means "all addresses", and will resolve into ipv4 0.0.0.0 + ipv6 ::
Then - bind to 0.0.0.0 will succeed, and binding to :: might then fail (EADDRINUSE), if dual-stack option is on (aka bind to ipv4 addr, when binding to ipv6 addr). Details here: https://stackoverflow.com/questions/51911564/cannot-bind-socket-to-port-in-both-ipv4-and-ipv6
So to avoid dependency on kernel parameter - we just turn dual stack off explicitly.

This might not be the best way (have not checked how nginx or mysql do it to be fair), but it works.

Reply:
checked man for setsockopt and changed int(~0) to 1, this indeed should work and is simpler.